src/Acuminator/Acuminator.Analyzers/Analyzers/Event/ConnectionScopeInRowSelectingAnalyzer.cs

```diff
@@ -0,0 +1,166 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Acuminator.Utilities;
+using Acuminator.Utils.RoslynExtensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Acuminator.Analyzers
+{
+	[DiagnosticAnalyzer(LanguageNames.CSharp)]
+	public class ConnectionScopeInRowSelectingAnalyzer : PXDiagnosticAnalyzer
+	{
+		private class Walker : NestedInvocationWalker
+		{
+			private class PXConnectionScopeVisitor : CSharpSyntaxVisitor<bool>
+			{
+				private readonly PXContext _pxContext;
+				private readonly SemanticModel _semanticModel;
+
+				public PXConnectionScopeVisitor(PXContext pxContext, SemanticModel semanticModel)
+				{
+					pxContext.ThrowOnNull(nameof(pxContext));
+					semanticModel.ThrowOnNull(nameof(semanticModel));
+					
+					_pxContext = pxContext;
+					_semanticModel = semanticModel;
+				}
+
+				public override bool VisitUsingStatement(UsingStatementSyntax node)
+				{
+					return (node.Declaration?.Accept(this) ?? false) || (node.Expression?.Accept(this) ?? false);
+				}
+
+				public override bool VisitObjectCreationExpression(ObjectCreationExpressionSyntax node)
+				{
+					var symbolInfo = _semanticModel.GetSymbolInfo(node.Type);
+					return symbolInfo.Symbol?.OriginalDefinition != null 
+						&& symbolInfo.Symbol.OriginalDefinition.Equals(_pxContext.PXConnectionScope);
+				}
+			}
+
+			private static readonly IEnumerable<string> MethodPrefixes = new[] { "Select", "Search" };
```

Comment:
@SENya1990 we also have Yaql and many other things, but there is no point to use them in RowSelecting, so they are not analyzed.

Reply:
@dropsonic Ithink it is a matter of supported DB access types - if we deicde to analyze PXDatabase then it is strange to not analyze all of its methods but just a subset.
PXUpdate - family of methods is still a public part of BQL so we have to support it too.
This is not so big change since the PXUpdate methods are already considered in the IsBqlCommand helper method + add two more strings to the `MethodPrefixes `